README.md

```diff
@@ -2,20 +2,118 @@
+├─ user
+|   ├─ __snapshots__
+|   |     └─ index.spec.js.snap 
+|   └─ index.spec.js
+├─ job
+¦   └─ index.spec.js
+¦
+¦
+├─ _xoWithTestHelpers.js
+└─ util.js
+```
 
+The tests can describe xo methods or scenarios:  
 ```
-> npm ci
-> npm test
+describe("user", () => {
+
+  // testing a method
+  describe(".set()", () => {
+    it("sets an email", async () => {
+      // some tests using xo methods and helpers from _xoWithTestHelpers.js
+      id = await xo.createUser(SIMPLE_USER);
+      expect(await xo.call("user.set", params)).toBe(true);
+      expect(omit(await xo.getUser(id), "id")).toMatchSnapshot();
+    });
+  });
+      
+  // testing a scenario
+  test("create two users, modify a user email to be the same with the other and fail trying to connect them", () => {
+    /* some tests */
+  });
+
+});
 ```
 
+### Best practices
+
+- The test environment must remain the same before and after each test:  
+  * each resource created must be deleted  
+  * existing resources should not be altered
+
+- Make a sentence for the title of the test. It must be clear and consistent.
+
+- If the feature you want to test is not implemented : write it and skip it, using `it.skip()`.
+
+- Take values ​​that cover the maximum of testing possibilities.
+
+- If you make tests which keep track of large object, it is better to use snapshots.  
+
+- `_xoWithTestHelpers.js` contains helpers to create temporary resources and to interface with XO.  
+  You can use it if you need to create resources which will be automatically deleted after the test:  
+  ```
+  import xo from "util"
+
+  describe(".create()", () => {
+    it("creates a user without permission", async () => {
+      // The user will be deleted automatically at the end of the test
+      const userId = await xo.createUser({
+        email: "wayne1@vates.fr",
+        password: "batman1",
+      });
+      expect(omit(await xo.getUser(userId), "id")).toMatchSnapshot();
+    });
+  });
+  ```
+
+  The available helpers:  
+  * `createUser(params)`  
+  * `deleteAllUsers()`  
+  * `getUser(id)`  
+  * `createTempJob(jobId)`  
+  * `deleteTempJobs()`  
+  * `createTempBackupNgJob(params)`  
+  * `deleteTempBackupNgJob()`  
+
+## Usage
+
+- To run the tests:  
+  ```
+  > npm ci
+  > yarn test
+  ```
+
+  You get all the test suites passed (`PASS`) or failed (`FAIL`).  
+  ```
+  > yarn test
+  yarn run v1.9.4
+  $ jest
+   PASS  src/user/user.spec.js
+   PASS  src/job/job.spec.js
+   PASS  src/backupNg/backupNg.spec.js
+
+  Test Suites: 3 passed, 3 total
+  Tests:       2 skipped, 36 passed, 38 total
+  Snapshots:   35 passed, 35 total
+  Time:        7.257s, estimated 8s
+  Ran all test suites.
+  Done in 7.92s.
+  ```
+
+- You can display individual test results with the test suite hierarchy by using: `> yarn test --verbose`  
+
+- Update snapshots: `> yarn test -u`. ⚠ Be careful to not impact the existing tests!  
+  You can see how many and which snapshots have been written, removed or updated.  
```

Comment:
IMHO you can remove this, it's better (and easier) to update the snapshots from the watch mode, do you agree?

Reply:
You're right. I am going to remove it :+1: